Source/engine/palette.cpp

```diff
@@ -150,28 +135,13 @@ void CycleColors(int from, int to)
  */
 void CycleColorsReverse(int from, int to)
 {
-	{
-		SDL_Color col = system_palette[to];
-		for (int i = to; i > from; i--) {
-			system_palette[i] = system_palette[i - 1];
-		}
-		system_palette[from] = col;
-	}
+	std::rotate(system_palette.begin() + from + 1, system_palette.begin() + from, system_palette.begin() + to);

```

Comment:
Is this actually cycling them in the right to left direction?

Reply:
yes